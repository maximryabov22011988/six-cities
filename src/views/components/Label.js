import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  isShow: PropTypes.bool.isRequired,
  name: PropTypes.string,
  parentClassName: PropTypes.string
};

function Label({ isShow, name, parentClassName }) {
  return (
    isShow && (
      <div className={`${parentClassName}__mark`}>
        <span>{name}</span>
      </div>
    )
  );
}

Label.propTypes = propTypes;

export default Label;
