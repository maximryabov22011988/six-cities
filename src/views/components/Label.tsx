import * as React from 'react';

interface Props {
  isShow: boolean,
  name: string,
  parentClassName: string,
}

function Label({ isShow, name, parentClassName }: Props) {
  return (
    isShow && (
      <div className={`${parentClassName}__mark`}>
        <span>{name}</span>
      </div>
    )
  );
}

export default Label;
