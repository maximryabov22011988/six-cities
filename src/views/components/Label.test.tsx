import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Label from './Label';

it('Label correctly render', () => {
  const tree = renderer
    .create(
      <Label isShow={true} name="Premium" parentClassName="place-card" />
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
