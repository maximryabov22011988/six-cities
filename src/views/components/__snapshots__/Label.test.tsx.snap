// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label correctly render 1`] = `
<div
  className="place-card__mark"
>
  <span>
    Premium
  </span>
</div>
`;
